compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/StructuredGraph.java

```diff
@@ -978,6 +980,14 @@ public void setAfterExpandLogic() {
         isAfterExpandLogic = true;
     }
 
+    public boolean isAfterReassociation() {
+        return isAfterReassociation;
+    }
+
+    public void setAfterReassociation() {
```

Comment:
In the course of my suggestion for the mul node this can be removed in favor of the shift to mul transformation